components/libraryprovider.py

```diff
@@ -190,6 +192,11 @@ def validate_library(yaml_contents, yaml_path):
         else:
             validated_library['repo_url'] = None
 
+        if 'vendoring' in library and 'patches' in library['vendoring']:
+            validated_library['has_patches'] = not not library['vendoring']['patches']
```

Comment:
oh dang, `not not` for turning a variable's value into a true/false value is neat